File: bintree_csharp/binarytree/traversal/ITraversal.cs
using System;
using System.Collections.Generic;


namespace my.binarytree.traversal
{
    interface ITraversal<TKey, TNode>
        where TKey : IComparable where TNode : BinNode<TKey, TNode>, new()
    {
        List<TKey> Traverse(TNode root, OrderTraversal order);
    }
}

File: bintree_java/src/binarytree/traversal/ITraversal.java
package binarytree.traversal;


import binarytree.BinNode;
import java.util.List;



public interface ITraversal< TKey extends Comparable<? super TKey>,
                                     TNode extends BinNode<TKey, TNode>
                                   >
{
    public abstract List<TKey> traverse(TNode root, OrderTraversal order);
}
